Serialization of Themes

I see that Widgets have code for serialization using the Serializer (core.h/ serializer.cpp).
For me, I'd like create Themes for buttons, text box styles and save them.

I could go two ways:  1) save the themes in JSON, read them and create Themes on the fly, or 2) use the serializer to save/load themes, which I think stores in binary format.

Any comments or suggestions on the best way to approach this?

A little follow-up:  When I tried to add use of the **Serializer** in example2.cpp, I get a linking error. Which is strange since these methods are both defined and _CMakeLists_ file is including _serializer/core.h_ and _src/serializer.cpp_ :

```
Undefined symbols for architecture x86_64:
  "nanogui::Serializer::Serializer(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool)", referenced from:
      _main in 0.o
  "nanogui::Serializer::~Serializer()", referenced from:
      _main in 0.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/include/nanogui/serializer/core.h
+++ b/include/nanogui/serializer/core.h
@@ -53,7 +53,7 @@ NAMESPACE_END(detail)
  * ``nanogui/serializer/opengl.h``, and ``nanogui/serializer/sparse.h`` must
  * be included to serialize the respective data types.
  */
-class Serializer {
+class NANOGUI_EXPORT Serializer {
 protected:
 // this friendship breaks the documentation
 #ifndef DOXYGEN_SHOULD_SKIP_THIS

